### PR TITLE
Release 0.0.1

### PR DESCRIPTION
**Pull request by Evgenia-bit**
## Changes
- Implemented functionality that allows you to create a fading text in case of overflow.
- Added example.
- Package is ready for release.

Demonstration:

Screenshot 2024-02-06 at 16 59 58
Screenshot 2024-02-06 at 17 00 42

## Checklist for self-check
- [x] Commits and PRs have been filed according to [the rules on the project](https://github.com/surfstudio/surf-flutter-app-template#workflow-in-a-repository).
- [x] The author is marked as an assigne and assigned mandatory reviewers.
- [x] Required labels marked
- [ ] Specified related tasks and/or related PRs.
- [x] Specified Changes.
- [x] Attached videos/screenshots demonstrating the fix/feature.
- [x] All unspecified fields in the PR description deleted.
- [x] New code covered by tests.

## Checklist for reviewers
- [ ] CI passed successfully _(with a green check mark)_.
- [ ] PR is atomic, by volume no more than 400 (+-) corrected lines (not including codogen).

Design:
- [ ] System design corresponds to the agreements on structure and architecture on the project.
- [ ] The code is decomposed into necessary and sufficient components.

Functionality:
- [ ] The code solves the problem.
- [ ] Any changes to the user interface are reasonable and look good.

Complexity:
- [ ] The code is clear, easy to read, functions are small, no more than 50 lines.
- [ ] The logic is not overcomplicated, there is no overengineering (no code sections that may be needed in the future, but no one knows about it).

Tests:
- [ ] Updated or added tests for mandatory components.
- [ ] The tests are correct, helpful, and well designed/developed.

Naming:
- [ ] The naming of variables, methods, classes and other components is understandable.

Comments:
- [ ] The comments are understandable and helpful.

Documentation:
- [ ] All labels are correct
- [ ] Technical documentation updated (after approval, updates last reviewer).

### CODE REVIEW

File: lib/faded_text.dart [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+  }
+
+  @override
+  bool shouldRepaint(covariant CustomPainter oldDelegate) {
+    return false;
+  }
+
+  /// This method is used to create a gradient shader to fade the text.
+  Shader? _overflowShader() {
+    textPainter.layout(maxWidth: constraints.maxWidth);
```
Comment (maxim-sysoev) [inline lib/faded_text.dart]: Is it necessarily to use `constraints`? Cannot use `size` passed to paint?

Comment (Evgenia-bit) [reply lib/faded_text.dart]: Yes, it can be done via `size`

File: .github/ISSUE_TEMPLATE/documentation_update.md [hunk 2]
```diff
@@ -0,0 +1,24 @@
+---
+name: Documentation update
+about: Fixing a problem or improve in an existing docs page.
+labels: documentation
+assignees: MbIXjkee
```
Comment (internetova) [inline .github/ISSUE_TEMPLATE/documentation_update.md]: This should be deleted

File: .github/ISSUE_TEMPLATE/feature_request.md [hunk 3]
```diff
@@ -0,0 +1,24 @@
+---
+name: Feature Request
+about: I'd like Faded Text to do something new.
+labels: enhancement
+assignees: MbIXjkee
```
Comment (internetova) [inline .github/ISSUE_TEMPLATE/feature_request.md]: This should be deleted

File: .github/PULL_REQUEST_TEMPLATE/bug_fix.md [hunk 4]
```diff
@@ -0,0 +1,29 @@
+---
+name: Bug fix
+about: Fixing a problem with Faded Text.
+assignees: MbIXjkee
```
Comment (internetova) [inline .github/PULL_REQUEST_TEMPLATE/bug_fix.md]: This should be deleted

File: .github/PULL_REQUEST_TEMPLATE/documentation_update.md [hunk 5]
```diff
@@ -0,0 +1,29 @@
+---
+name: Documentation update
+about: Fixing a problem or improve in an existing docs page.
+labels: documentation
+assignees: MbIXjkee
```
Comment (internetova) [inline .github/PULL_REQUEST_TEMPLATE/documentation_update.md]: This should be deleted

File: .github/PULL_REQUEST_TEMPLATE/new_feature.md [hunk 6]
```diff
@@ -0,0 +1,27 @@
+---
+name: New feature
+labels: enhancement
+assignees: MbIXjkee
```
Comment (internetova) [inline .github/PULL_REQUEST_TEMPLATE/new_feature.md]: This should be deleted

**Review by maxim-sysoev — APPROVED** [review]

**Pull request merged**